arangod/Aql/Functions.cpp

```diff
@@ -4484,6 +4484,43 @@ AqlValue Functions::Average(ExpressionContext* expressionContext, transaction::M
   return AqlValue(AqlValueHintNull());
 }
 
+/// @brief function PRODUCT
+AqlValue Functions::Product(ExpressionContext* expressionContext, transaction::Methods* trx,
+                            VPackFunctionParameters const& parameters) {
+  AqlValue const& value = extractFunctionParameterValue(parameters, 0);
+
+  if (!value.isArray()) {
+    // not an array
+    registerWarning(expressionContext, "PRODUCT", TRI_ERROR_QUERY_ARRAY_EXPECTED);
+    return AqlValue(AqlValueHintNull());
+  }
+
+  AqlValueMaterializer materializer(trx);
+  VPackSlice slice = materializer.slice(value, false);
+  double product = 0.0;
+  bool seen = false;
+  for (VPackSlice it : VPackArrayIterator(slice)) {
+    if (it.isNull()) {
+      continue;
+    }
+    if (!it.isNumber()) {
+      return AqlValue(AqlValueHintNull());
+    }
+    double const number = it.getNumericValue<double>();
+
+    if (!std::isnan(number) && number != HUGE_VAL && number != -HUGE_VAL) {
+      if (!seen) {
+        seen = true;
+        product = number;
+      } else {
+        product *= number;
+      }
+    }
+  }
+
+  return ::numberValue(product, false);
```

Comment:
This is somewhat odd because it returns `0.0` for empty arrays. In math a empty product is `1.0`. This would also get rid of `seen`.

Reply:
Ok, I wasn't aware of that.
Good that mathematicians check PRs :eyes: !

Reply:
done